### DISCUSSION TITLE
GPS modules problem

### DISCUSSION THREAD

**Discussion by Aripaule** [AllskyTeam/allsky-modules#220] [Q&A]
About GPS modules in allsky:

I have two models:

https://pt.aliexpress.com/item/1005007477622080.html?spm=a2g0o.order_list.order_list_main.119.7bb8caa4ZyNvI9&gatewayAdapt=glo2bra

https://pt.aliexpress.com/item/1005006787427660.html?spm=a2g0o.order_list.order_list_main.434.7bb8caa4ZyNvI9&gatewayAdapt=glo2bra

The gps is connected via serial output, and Vin is connected to the pi, 3V (I also tested 5V but it doesn't seem to have fix anything)

1) I installed gpsd
2) I activated the serial port in raspi-config
3) I edited the file for multiple ID’S: /etc/default/gpsd 
DEVICES=/dev/serial0
DEVICES=/dev/seria11
DEVICES=/dev/TTYAM0
DEVICES=/dev/TTYS0
DEVICES=/dev/TTYAM1
After each test, I restarted the GPSD.
4) I also tried with the module outside the house (to improve the signal)

The vast majority have no effect, and the output generated by cgps is something like this:

![image](https://github.com/user-attachments/assets/ff08e098-bf81-4b52-ae9c-5890c1030e67)

5) the only command that gave a different result was DEVICES=/dev/TTYS0 (with GY-NEO6M module)

![image](https://github.com/user-attachments/assets/2213df1b-ed33-47ec-b5d9-e0623fe0abd5)

However, the signal did not fix at any time, even after more than 10 minutes of waiting.

If anyone has any ideas on what can be done, I would appreciate it.

Ari.

**Comment by Alex-developer** [comment]
First off when you plug the gps in what device appears in /dev ?

**Reply by Aripaule** [reply]
![image](https://github.com/user-attachments/assets/a64394ae-ea0b-4808-aff3-475c57a0593c)

**Comment by Alex-developer** [comment]
Its a bit beyond the scope of allsky but I will try and help where i can

I would install a program to read the serial port directly and see what data is being returned

```
sudo apt install minicom
minicom -D /dev/ttyAMA0 -b 9600
```

You may have to alter the baud rate depending upon your GPS module

**Reply by Aripaule** [reply]
ok, thank you so much!

![image](https://github.com/user-attachments/assets/adba03bc-21ae-400a-b20c-3375293c9fe9)

**Reply by Alex-developer** [reply]
> minicom -D /dev/ttyAMA0 -b 9600

Looks like you are not getting any dta from the gps, I get soemthing like the following

```
minicom -D /dev/ttyACM0 -b 9600

Welcome to minicom 2.8

OPTIONS: I18n
Port /dev/ttyACM0, 20:28:55

Press CTRL-A Z for help on special keys

$GPTXT,01,01,02,u-blox ag - www.u-blox.com*50
$GPTXT,01,01,02,HW  UBX-G70xx   00070000 FF7FFFFFo*69
$GPTXT,01,01,02,ROM CORE 1.00 (59842) Jun 27 2012 17:43:52*59
$GPTXT,01,01,02,PROTVER 14.00*1E
$GPTXT,01,01,02,ANTSUPERV=AC SD PDoS SR*20
$GPTXT,01,01,02,ANTSTATUS=OK*3B
$GPTXT,01,01,02,LLC FFFFFFFF-FFFFFFFF-FFFFFFFF-FFFFFFFF-FFFFFFFD*2C
$GPRMC,,V,,,,,,,,,,N*53
$GPVTG,,,,,,,,,N*30
$GPGGA,,,,,,0,00,99.99,,,,,,*48
$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30
$GPGSV,1,1,02,11,,,21,12,,,21*78
$GPGLL,,,,,,V,N*64
$GPRMC,,V,,,,,,,,,,N*53
$GPVTG,,,,,,,,,N*30
$GPGGA,,,,,,0,00,99.99,,,,,,*48
$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30
$GPGSV,1,1,02,11,,,21,12,,,21*78
$GPGLL,,,,,,V,N*64
$GPRMC,,V,,,,,,,,,,N*53
$GPVTG,,,,,,,,,N*30
```
You might want to check the baud rate for the gps module you have and also the wiring to make sure its correct


P.S My module is indoors so will never get a lock !

Alex

**Comment by chvvkumar** [comment]
I have had terrible luck with the cheap GPS devices. The issue I narrowed down to is that the built-in battery backup (if they even have one) dies and the GPS has to do a cold boot every time you plug in. Even the expensive Sparkfun NEO-M9N I got to replace the NEO-7 modules is tempermental some times.
 
The N7 modules I have sometimes take hours to get a lock outside in full view of the sky. Try leaving it plugged in for a day and check back to see if it gets a lock.

Otherwise, my suggestion would be to get the u-blox ucenter software for windows and use a PC to see what's going on and if the module works in the first place.

Also remember most of the cheap GPS modules are not genuine u-box modules.

**Reply by Aripaule** [reply]
I understand, I didn't make any progress here. I think I'll leave the GPS module aside for now.

**Comment by Alex-developer** [comment]
If you are going to leave it for now I will close this discussion but feel free to open it again if you need any more help

Alex
